### PR TITLE
issues 153 and 150

### PR DESCRIPTION
**Pull request by l1ll1**
Hi,
I believe this PR fixes issues 153 (slurmdb_jobs.get missing jobs) and 150 (slurmdb_jobs.get to take a list of users).

Apparently job_cond.db_flags needs to default to SLURMDB_JOB_FLAG_NOTSET (found this in the sacct source code)

There appear some to be some missing fields on the slurmdb_job_rec_t which I fixed as well.

The userids need to be either the uidNumber as a string or the uidNumber as an int. Left as an exercise for the user to convert usernames to uid Numbers.

Happy to review the MR and add documentation if necessary.
Thanks for a great tool.

### CODE REVIEW

**Comment by giovtorres** [comment]
Thank you for the contribution, @l1ll1 :rocket: This looks great. Would you be able to write tests for this change to ensure this continues to work going forward?

Also, it's ok if some struct members aren't re-declared if they are not used.

**Comment by l1ll1** [comment]
Thanks @giovtorres 
I've added some checks. Hopefully they fit what you expect (the assume an existing slurm accounting database with some actual data it in)

regarding the struct members, would you like me to remove them? I've never worked in Cython and didn't realize it wasn't necessary to declare all members of the struct, so I found it a bit confusing ;-). I can cancel the PR and clean up the code base if that helps?

**Comment by giovtorres** [comment]
> regarding the struct members, would you like me to remove them?

It's ok, you can leave them in for completeness. Having them in won't affect the behaviour.

**Comment by giovtorres** [comment]
Hmm, it doesn't look like these tests executed.

File: tests/test-slurmdb.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return len(jobs)
+
+def njobs_slurmdb_jobs_get_byuid(start,end,uid):
+    print(uid)
+    jobs = pyslurm.slurmdb_jobs().get(starttime=start.encode('utf-8'), endtime=end.encode('utf-8'),userids=[uid])
+    print('njobs by py slumr {}'.format(len(jobs)))
+    return len(jobs)
+
+def get_user():
+    import pwd
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: Let's move this `import` to the top of the file

File: tests/test-slurmdb.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        yield username.strip().decode(),uid.pw_uid
+
+    
+
+def test_slurmdb_jobs_get():
+    starttime = (datetime.datetime.now()-datetime.timedelta(days=2)).strftime("%Y-%m-%dT00:00:00")
+    endtime = (datetime.datetime.now()-datetime.timedelta(days=1)).strftime("%Y-%m-%dT00:00:00")
+    njobs_pyslurm = njobs_slurmdb_jobs_get(starttime,endtime)
+    njobs_sacct = njobs_sacct_jobs(starttime,endtime)
+    assert_equals(njobs_pyslurm,njobs_sacct)
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: Add a space after the comma

File: tests/test-slurmdb.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def test_slurmdb_jobs_get_byuser():
+    userlist = list(get_user())
+    for user in userlist[:10]:
+        starttime = (datetime.datetime.now()-datetime.timedelta(days=2)).strftime("%Y-%m-%dT00:00:00")
+        endtime = (datetime.datetime.now()-datetime.timedelta(days=1)).strftime("%Y-%m-%dT00:00:00")
+        njobs_pyslurm = njobs_slurmdb_jobs_get_byuid(starttime,endtime,int(user[1])) 
+        njobs_sacct = njobs_sacct_jobs_byuser(starttime,endtime,user[0])
+        print('njobs by sacct {}'.format(njobs_sacct))
+        assert_equals(njobs_pyslurm,njobs_sacct)
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: Add a space after the comma

File: tests/test-slurmdb.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import, unicode_literals
+
+import pyslurm
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: I would reorder these as Python builtin modules, 3rd party modules and custom modules, all separated by a space.

**Comment by giovtorres** [comment]
> I've added some checks. Hopefully they fit what you expect (the assume an existing slurm accounting database with some actual data it in)

There's some opportunity for refactoring and removing some code duplication. Otherwise, logic is good.

File: tests/test-slurmdb.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    users = subprocess.Popen(['squeue', '-O', 'username', '-h'],stdout=subprocess.PIPE,stderr=None).communicate()
+    for username in users[0].splitlines():
+        print(username.decode())
+        uid = pwd.getpwnam("{}".format(username.strip().decode()))
+        yield username.strip().decode(),uid.pw_uid
+
+    
+
+def test_slurmdb_jobs_get():
+    starttime = (datetime.datetime.now()-datetime.timedelta(days=2)).strftime("%Y-%m-%dT00:00:00")
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: Could you add a doctstring here similar to the other test methods in this suite?

File: tests/test-slurmdb.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    endtime = (datetime.datetime.now()-datetime.timedelta(days=1)).strftime("%Y-%m-%dT00:00:00")
+    njobs_pyslurm = njobs_slurmdb_jobs_get(starttime,endtime)
+    njobs_sacct = njobs_sacct_jobs(starttime,endtime)
+    assert_equals(njobs_pyslurm,njobs_sacct)
+
+def test_slurmdb_jobs_get_byuser():
+    userlist = list(get_user())
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: Same here.... Could you add a doctstring here similar to the other test methods in this suite?

File: tests/test-slurmdb.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import pyslurm
+import subprocess
+from nose.tools import assert_equals, assert_true
+from socket import gethostname
+import datetime
+
+def njobs_sacct_jobs(start, end):
+    sacct = subprocess.Popen(['sacct','-S',start,'-E',end,'-n','-X','-a'],stdout=subprocess.PIPE,stderr=None).communicate()
+    return len(sacct[0].splitlines())
+
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: There should be two spaces between methods. Same applies below...

**Comment by l1ll1** [comment]
Thanks @giovtorres ... feeling suitably embarrassed: perhaps one day I'll get around to turning on pep8 checking right in vim and I won't submit such a dogs breakfast for PRs anymore ;-)

File: tests/test-slurmdb.py [hunk 8]
```diff
@@ -1,54 +1,81 @@
-from __future__ import absolute_import, unicode_literals
+import datetime
+import pwd
```
Comment (giovtorres) [inline tests/test-slurmdb.py]: :clap:

**Pull request merged**